drivers/mspi/mspi_stm32_ospi.c

```diff
@@ -0,0 +1,1823 @@
+/*
+ * Copyright (c) 2025 EXALT Technologies
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * **************************************************************************
+ * MSPI flash controller driver for stm32 series with multi-SPI peripherals
+ * This driver is based on the stm32Cube HAL OSPI driver
+ * with one mspi DTS node.
+ * **************************************************************************
+ */
+#define DT_DRV_COMPAT st_stm32_ospi_controller
+
+#include <zephyr/cache.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/drivers/mspi.h>
+#include <zephyr/drivers/dma/dma_stm32.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/irq.h>
+#include <zephyr/logging/log.h>
+#include <stm32_bitops.h>
+#include <stm32_ll_dma.h>
+#include "mspi_stm32.h"
+
+#define DT_OSPI_IO_PORT_PROP_OR(prop, default_value, index)                             \
+	COND_CODE_1(DT_NODE_HAS_PROP(DT_DRV_INST(index), prop),				\
+		    (_CONCAT(HAL_OSPIM_, DT_STRING_TOKEN(DT_DRV_INST(index), prop))),	\
+		    (default_value))
+
+#define DT_OSPI_PROP_OR(prop, default_value, index)                                     \
+	DT_PROP_OR(DT_DRV_INST(index), prop, default_value)
+
+LOG_MODULE_REGISTER(ospi_stm32, CONFIG_MSPI_LOG_LEVEL);
+
+static int mspi_stm32_ospi_context_lock(struct mspi_stm32_context *ctx,
+					const struct mspi_dev_id *req, const struct mspi_xfer *xfer,
+					bool lockon)
+{
+	if (k_sem_take(&ctx->lock, K_MSEC(xfer->timeout)) < 0) {
```

Comment:
```suggestion
	if (lockon && k_sem_take(&ctx->lock, K_MSEC(xfer->timeout)) < 0) {
```
or alternatively, `ARG_UNUSED(lockon);` or remove `lockon` argument if not to be used.

Ditto for `mspi_context_lock()` in **mspi_stm32_qspi.c** and `mspi_stm32_xspi_context_lock()` in **mspi_stm32_xspi.c**.